polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts

```diff
@@ -0,0 +1,314 @@
+
+export default function replaceBorderDeclarations(
+  fileInfo: FileInfo,
+  _: API,
+  options: Options,
+) {
+  return postcss(plugin(options)).process(fileInfo.source, {
+    syntax: require('postcss-scss'),
+  }).css;
+}
+
+const processed = Symbol('processed');
+
+interface PluginOptions extends Options, NamespaceOptions {}
+
+const plugin = (options: PluginOptions = {}): Plugin => {
+  const namespacedRem = namespace('rem', options);
+  const namespacedBorder = namespace('border', options);
+  const namespacedBorderWidth = namespace('border-width', options);
+  const namespacedBorderRadius = namespace('border-radius', options);
+
+  return {
+    postcssPlugin: 'replace-sass-border',
+    Declaration(decl) {
+      // @ts-expect-error - Skip if processed so we don't process it again
+      if (decl[processed]) return;
+
+      if (!borderProps.has(decl.prop)) return;
+
+      /**
+       * A collection of transformable values to migrate (e.g. decl lengths, functions, etc.)
+       *
+       * Note: This is evaluated at the end of each visitor execution to determine whether
+       * or not to replace the declaration or insert a comment.
+       */
+      const targets: {replaced: boolean}[] = [];
+      let hasNumericOperator = false;
+      const parsedValue = valueParser(decl.value);
+
+      handleBorderProps();
+
+      if (targets.some(({replaced}) => !replaced || hasNumericOperator)) {
+        // Insert comment if the declaration value contains calculations
+        decl.before(
+          createInlineComment(POLARIS_MIGRATOR_COMMENT, {prose: true}),
+        );
+        decl.before(
+          createInlineComment(`${decl.prop}: ${parsedValue.toString()};`),
+        );
+      } else {
+        decl.value = parsedValue.toString();
+      }
+
+      //
+      // Handlers
+      //
+
+      function handleBorderProps() {
+        parsedValue.walk((node) => {
+          if (node.type === 'word') {
+            if (globalValues.has(node.value)) return;
+            if (isNumericOperator(node)) {
+              hasNumericOperator = true;
+              return;
+            }
+
+            const dimension = valueParser.unit(node.value);
+
+            if (!isTransformableLength(dimension)) return;
+
+            targets.push({replaced: false});
+
+            const valueInPx = toTransformablePx(node.value);
+
+            if (decl.prop === 'border' || decl.prop.includes('border-width')) {
+              if (!isKeyOf(borderWidthLengthMap, valueInPx)) {
+                return;
+              }
+
+              node.value = `var(${borderWidthLengthMap[valueInPx]})`;
+            }
+
+            if (decl.prop.includes('radius')) {
+              if (!isKeyOf(borderRadiusLengthMap, valueInPx)) {
+                return;
+              }
+
+              node.value = `var(${borderRadiusLengthMap[valueInPx]})`;
+            }
+
+            targets.at(-1)!.replaced = true;
+
+            return;
+          }
```

Comment:
This covers "the happy path" (hard coded `px`, `rem`) for both `--p-border-width-X` and `--p-border-radius-X` token mapping.